### ISSUE TITLE
shipcloud

### ISSUE THREAD

**Issue by jverce** [PipedreamHQ/pipedream#3866]
### App
shipcloud

### Triggers
* New Shipment Status

### Actions
* Create Shipment
* Create Shipment Quote
* Find Shipment

**Comment by GTFalcao** [comment]
@jverce regarding the 'New Shipment Status' trigger, Shipcloud's API Docs list these event types:
`*, shipment.*, shipment.status.*, shipment.status.deleted, shipment.tracking.*, shipment.tracking.awaits_pickup_by_receiver, shipment.tracking.canceled, shipment.tracking.delayed, shipment.tracking.delivered, shipment.tracking.destroyed, shipment.tracking.exception, shipment.tracking.label_created, shipment.tracking.not_delivered, shipment.tracking.notification, shipment.tracking.out_for_delivery, shipment.tracking.picked_up, shipment.tracking.transit, shipment.tracking.unknown`

In this context I think a new shipment status would be any of these tracking-related events, can you confirm if that's what you had in mind as well?

**Comment by jverce** [comment]
> @jverce regarding the 'New Shipment Status' trigger, Shipcloud's API Docs list these event types:
> 
> ```
> *, shipment.*, shipment.status.*, shipment.status.deleted, shipment.tracking.*, shipment.tracking.awaits_pickup_by_receiver, shipment.tracking.canceled, shipment.tracking.delayed, shipment.tracking.delivered, shipment.tracking.destroyed, shipment.tracking.exception, shipment.tracking.label_created, shipment.tracking.not_delivered, shipment.tracking.notification, shipment.tracking.out_for_delivery, shipment.tracking.picked_up, shipment.tracking.transit, shipment.tracking.unknown
> ```
> 
> In this context I think a new shipment status would be any of these tracking-related events, can you confirm if that's what you had in mind as well?

@GTFalcao I agree, seems like all these webhook events refer to a status change in a shipment.
The good thing about this API IMHO is that we can use wildcards, so we could easily provide 3 event sources for this purpose:
1. Shipment status updates (e.g. if the shipment record has been deleted). This event source would react to the `shipment.status.*` webhook events
2. Shipment tracking status updates (which would make more sense if an e-commerce shop wants to keep their customers posted regarding their packages for example). The webhook events for this one would be `shipment.tracking.*`
3. A catch-all event source that reacts to `*` (i.e. any/all webhook events)

Does that make sense to you?

**Comment by GTFalcao** [comment]
> > @jverce regarding the 'New Shipment Status' trigger, Shipcloud's API Docs list these event types:
> > ```
> > *, shipment.*, shipment.status.*, shipment.status.deleted, shipment.tracking.*, shipment.tracking.awaits_pickup_by_receiver, shipment.tracking.canceled, shipment.tracking.delayed, shipment.tracking.delivered, shipment.tracking.destroyed, shipment.tracking.exception, shipment.tracking.label_created, shipment.tracking.not_delivered, shipment.tracking.notification, shipment.tracking.out_for_delivery, shipment.tracking.picked_up, shipment.tracking.transit, shipment.tracking.unknown
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In this context I think a new shipment status would be any of these tracking-related events, can you confirm if that's what you had in mind as well?
> 
> @GTFalcao I agree, seems like all these webhook events refer to a status change in a shipment. The good thing about this API IMHO is that we can use wildcards, so we could easily provide 3 event sources for this purpose:
> 
> 1. Shipment status updates (e.g. if the shipment record has been deleted). This event source would react to the `shipment.status.*` webhook events
> 2. Shipment tracking status updates (which would make more sense if an e-commerce shop wants to keep their customers posted regarding their packages for example). The webhook events for this one would be `shipment.tracking.*`
> 3. A catch-all event source that reacts to `*` (i.e. any/all webhook events)
> 
> Does that make sense to you?
I've currently implemented a single source with an `event type` prop that lets the user choose exactly which events they want the source to listen to, out of those listed above.

We can easily split this into the three sources that you proposed though, it's a fairly small code change.
